Emby.Dlna/Didl/DidlBuilder.cs

```diff
@@ -425,57 +425,99 @@ private string GetDisplayName(BaseItem item, StubType? itemStubType, BaseItem co
                 }
             }
 
-            if (item is Episode episode && context is Season season)
+            return item is Episode episode
+                ? GetEpisodeDisplayName(episode, context)
+                : item.Name;
+        }
+
+        /// <summary>
+        /// Gets episode display name appropriate for the given context.
+        /// </summary>
+        /// <remarks>
+        /// If context is a season, this will return a string containing just episode number and name.
+        /// Otherwise the result will include series nams and season number.
+        /// </remarks>
+        /// <param name="episode">The episode.</param>
+        /// <param name="context">Current context.</param>
+        /// <returns>Formatted name of the episode.</returns>
+        private string GetEpisodeDisplayName(Episode episode, BaseItem context)
+        {
+            string[] components;
+
+            if (context is Season season)
             {
                 // This is a special embedded within a season
-                if (item.ParentIndexNumber.HasValue && item.ParentIndexNumber.Value == 0
+                if (episode.ParentIndexNumber.HasValue && episode.ParentIndexNumber.Value == 0
                     && season.IndexNumber.HasValue && season.IndexNumber.Value != 0)
                 {
                     return string.Format(
                         CultureInfo.InvariantCulture,
                         _localization.GetLocalizedString("ValueSpecialEpisodeName"),
-                        item.Name);
+                        episode.Name);
                 }
 
-                if (item.IndexNumber.HasValue)
-                {
-                    var number = item.IndexNumber.Value.ToString("00", CultureInfo.InvariantCulture);
+                // inside a season use simple format (ex. '12 - Episode Name')
+                var epNumberName = GetEpisodeIndexFullName(episode);
+                components = new[] { epNumberName, episode.Name };
+            }
+            else
+            {
+                // outside a season include series and season details (ex. 'TV Show - S05E11 - Episode Name')
+                var epNumberName = GetEpisodeNumberDisplayName(episode);
+                components = new[] { episode.SeriesName, epNumberName, episode.Name };
+            }
 
-                    if (episode.IndexNumberEnd.HasValue)
-                    {
-                        number += "-" + episode.IndexNumberEnd.Value.ToString("00", CultureInfo.InvariantCulture);
-                    }
+            return string.Join(" - ", components.Where(NotNullOrWhiteSpace));
```

Comment:
why didn't you use a lambda function here?..
```suggestion
            return string.Join(" - ", components.Where(comp => !string.IsNullOrWhiteSpace(comp)));
```

Reply:
Just to make it look cleaner.